### ISSUE TITLE
Make tests work in offline environment without pytest-venv

### ISSUE THREAD

**Issue by frenzymadness** [thoth-station/micropipenv#108]
It would be nice to have a simple way to skip all the tests that require an internet connection (not available in the RPM build process) and make them work without pytest-venv which is not available in Fedora/RHEL.

The second enhancement is kinda questionable because I can simply package it for Fedora but I think that there are no many tests requiring pytest-venv and not requiring internet connection so there is no real benefit in having pytest-venv and skipping it because of the internet.

In other words, let's skip online tests first and see how many of the rest needs the venv fixture.

**Comment by fridex** [comment]
Related discussion: https://github.com/thoth-station/micropipenv/pull/107#issuecomment-663346365

**Comment by fridex** [comment]
BTW we can also bundle pytest-venv in the testsuite. It's very simple and straightforward, [see it's implementation](https://github.com/mmerickel/pytest-venv/blob/master/src/pytest_venv/__init__.py).

**Comment by frenzymadness** [comment]
How many tests are offline and need pytest-venv given the fact that we do testing only with the system pip and toml during the build of RPM?

**Comment by fridex** [comment]
> How many tests are offline and need pytest-venv given the fact that we do testing only with the system pip and toml during the build of RPM?

```console
$ grep '^def test_' tests/test_micropipenv.py | wc -l        # all tests
41
$ grep '^def test_.*(venv)' tests/test_micropipenv.py | wc -l
19
```

I think the venv can be created also for RHEL tests, without explicit pip installation but using the system pip. Hence bundling pytest-venv might be useful. But I haven't checked all the tests, and I'm not sure if there is any exception to consider.

**Comment by fridex** [comment]
> But I haven't checked all the tests, and I'm not sure if there is any exception to consider.

But maybe (all?) the venv tests require downloading packages from Internet, so they might not be useful for offline testing in a restricted environment.
